src/travel-aid/components/StopSignalButton.tsx

```diff
@@ -0,0 +1,98 @@
+import {TravelAidTexts} from '@atb/translations/screens/subscreens/TravelAid';
+import type {EstimatedCallWithQuayFragment} from '@atb/api/types/generated/fragments/estimated-calls';
+import {AUTHORITY} from '@env';
+import {addMinutes, isBetween} from '@atb/utils/date';
+import React from 'react';
+import type {ServiceJourneyWithGuaranteedCalls} from '@atb/travel-aid/types';
+import {View} from 'react-native';
+import {MessageInfoText} from '@atb/components/message-info-text';
+import {Confirm} from '@atb/assets/svg/mono-icons/actions';
+import type {MutationStatus} from '@tanstack/react-query';
+import type {SendStopSignalRequestType} from '@atb/api/stop-signal';
+
+export const StopSignalButton = ({
+  serviceJourney,
+  fromQuayId,
+  onPress,
+  status,
+}: {
+  serviceJourney: ServiceJourneyWithGuaranteedCalls;
+  fromQuayId?: string;
+  onPress: (args: SendStopSignalRequestType) => void;
+  status: MutationStatus;
+}) => {
+  const {isTravelAidStopButtonEnabled} = useFeatureToggles();
+  const {theme} = useTheme();
+  const {t} = useTranslation();
+  const styles = useStyles();
+
+  if (!isTravelAidStopButtonEnabled) return null;
+
+  const selectedCall = serviceJourney.estimatedCalls.reduce((selected, cur) =>
+    cur.quay.id === fromQuayId ? cur : selected,
+  );
+
+  const shouldShow = shouldShowStopButton(serviceJourney, selectedCall);
+  if (!shouldShow) return null;
+
+  const isEnabled = isStopButtonEnabled(selectedCall);
+
+  return (
+    <View>
+      <Button
+        interactiveColor={theme.color.interactive['0']}
+        text={t(
+          status === 'success'
+            ? TravelAidTexts.stopButton.textAfterSubmit
+            : TravelAidTexts.stopButton.text,
+        )}
+        onPress={() =>
+          onPress({
+            serviceJourneyId: serviceJourney.id,
+            quayId: selectedCall.quay.id,
+            date: selectedCall.date,
+          })
+        }
+        loading={status === 'loading'}
+        active={status === 'success'}
+        disabled={!isEnabled}
+        rightIcon={status === 'success' ? {svg: Confirm} : undefined}
+      />
+      {!isEnabled && (
+        <MessageInfoText
+          type="warning"
+          message={t(TravelAidTexts.stopButton.notAvailable)}
+          style={styles.notAvailableWarning}
+        />
+      )}
+    </View>
+  );
+};
+
+const shouldShowStopButton = (
+  serviceJourney: ServiceJourneyWithGuaranteedCalls,
+  call: EstimatedCallWithQuayFragment,
+) => {
+  return (
+    !call.actualDepartureTime &&
+    serviceJourney.transportMode &&
+    ['bus', 'tram'].includes(serviceJourney.transportMode) &&
+    AUTHORITY === serviceJourney.line.authority?.id
+  );
+};
+
+const isStopButtonEnabled = (call: EstimatedCallWithQuayFragment) => {
+  if (!call.realtime) return false;
+
+  const activationWindowStart = addMinutes(call.expectedArrivalTime, -60);
+  const activationWindowEnd = addMinutes(call.expectedArrivalTime, -2);
```

Comment:
I'm sure this has been discussed, but 2 minutes before sounds quite strict to me 🤔 
Is this something that should be remotely configurable?

Also do we have a max refresh frequency for `expectedArrivalTime`? If not, it is currently possible that the enabled state will jump back and forth rapidly.

Reply:
It will be configurable, as mentioned in the PR:
> - Config in firestore like applicable transport modes and minutes
before availability.

The service journey, including all estimated calls, is refetched every 10 second. And yeah, you are right, in theory it can jump back and forth from being disabled or not at this refetch. I don't think it will be such a big problem that it needs explicit handling, but might change my mind if we get some reports.

Reply:
Might not be an issue then :+1:
If it comes to it, we could consider using `now` from `useNow` instead of `new Date()`, and return `isEnabled` as a state variable that is only updated in a useEffect with now as a dependency..